tests/test_ilastik/test_workflows/test_HeadlessPixelClassificationWorkflow.py

```diff
@@ -224,3 +231,128 @@ def test_distributed_results_are_identical_to_single_process_results(
+                "version": "0.4",
+                "axes": [
+                    {"type": "space", "name": "c"},
+                    {"type": "space", "name": "y", "unit": "pixel"},
+                    {"type": "space", "name": "x", "unit": "pixel"},
+                ],
+                "datasets": [
+                    {
+                        "path": "s0",
+                        "coordinateTransformations": [
+                            {"scale": [1.0 for _ in dataset_shape], "type": "scale"},
+                            {"translation": [0.0 for _ in dataset_shape], "type": "translation"},
+                        ],
+                    },
+                    {
+                        "path": "s1",
+                        "coordinateTransformations": [
+                            {"scale": [2.0 for _ in scaled_shape], "type": "scale"},
+                            {"translation": [0.0 for _ in scaled_shape], "type": "translation"},
+                        ],
+                    },
+                ],
+                "coordinateTransformations": [],
+            }
+        ]
+    }
+    (zarr_dir / ".zattrs").write_text(json.dumps(zattrs))
+
+    image_original = numpy.random.randint(0, 256, dataset_shape, dtype=numpy.uint16)
+    image_scaled = image_original[:, ::2, ::2]
+    chunks = tuple(chunk_size)
+    zarr.array(image_original, chunks=chunks, store=zarr.DirectoryStore(str(zarr_dir / "s0")), **OME_ZARR_V_0_4_KWARGS)
+    zarr.array(image_scaled, chunks=chunks, store=zarr.DirectoryStore(str(zarr_dir / "s1")), **OME_ZARR_V_0_4_KWARGS)
+
+    return {"s0": image_original, "s1": image_scaled}
+
+
+def test_headless_from_ome_zarr_file_uri(testdir, tmp_path, pixel_classification_ilp_2d3c, ome_zarr_store_on_disc):
+    # Request raw scale to test that the full path is used.
+    # The loader implementation defaults to loading the lowest resolution (last scale).
+    raw_data_path = tmp_path / "some.zarr/s0"
+    output_path = tmp_path / "out_100x100y3c.h5"
+    run_headless_pixel_classification(
+        testdir,
+        project=pixel_classification_ilp_2d3c,
+        raw_data=raw_data_path.as_uri(),
+        output_filename_format=str(output_path),
+        ignore_training_axistags=True,
+    )
+
+    assert output_path.exists()
+    with h5py.File(output_path, "r") as f:
+        exported = f["exported_data"][()]
+        assert exported.shape == (2, 100, 100)
+        # Make sure the raw data was actually loaded (zarr fills chunks that fail to load with 0)
+        assert numpy.count_nonzero(exported) > 10000
+
+
+def wait_for_server(host_and_port: Tuple[str, int], timeout=5):
+    end_time = time.time() + timeout
+    while time.time() < end_time:
+        try:
+            with socket.create_connection(host_and_port, timeout=1):
+                return True
+        except OSError:
+            time.sleep(0.1)
+    raise RuntimeError(f"Server on {':'.join(host_and_port)} didn't start within {timeout} seconds")
```

Comment:
for fun I ran something on `localhost:8889`. At least on mac, this `wait_for_server` doesn't error out. It would be cool if this test wouldn't rely on a specific port being open (maybe random, check if available first?).

Reply:
True, trying random and checking availability would be sensible

Reply:
If you bind to port 0, then a random, free port will be selected automatically.